### PR TITLE
#47 - Remove GuzzleStack history

### LINKED ISSUE #47: Reduce memory usage of guzzle clients using BaseCategory::getGuzzleHandlerStack()
Currently for weird legacy debug reasons (??) BaseCategory::getGuzzleHandlerStack() stores all requests in memory using GuzzleHttp\Middleware::history(). For providers whose API calls return a LOT of data, this can lead to spikes in memory consumption during function execution.

Let's remove the guzzle history middleware.

### PR DESCRIPTION
**Pull request by RoussKS**
Closes #47 

- removed guzzle stack history middleware and relevant properties/methods. Updated CHANGELOG

### CODE REVIEW

**Comment by RoussKS** [comment]
Static analysis will be resolved once #60 is merged & we merge `dev-v5` to the current branch

File: src/Provider/BaseCategory.php [hunk 1]
```diff
@@ -123,73 +104,4 @@ final public function getSystemInfo(): SystemInfo
 
         return $this->systemInfo;
     }
-
-    /**
-     * Get a guzzle handler stack which logs requests/responses if provider is
-     * an instance of LogsDebugData and $debugLog === true. Requests and responses
-     * will also be stored in $this->guzzleHistory.
-     */
-    protected function getGuzzleHandlerStack(bool $debugLog = false): HandlerStack
```
Comment (uphlewis) [inline src/Provider/BaseCategory.php:132]: We need to keep this method for PSR-3 logging middleware as it's widely used

**Review by uphlewis — CHANGES_REQUESTED** [review]
We need to keep the logging middleware for guzzle and only need to remove the `$this->guzzleHistory` bits

File: src/Provider/BaseCategory.php (hunk 1 above)
Comment (RoussKS) [reply src/Provider/BaseCategory.php:132]: This is now updated. Static analysis will be fixed again when #60 is merged

**Pull request merged**